addons/sale_coupon/models/sale_coupon_program.py

```diff
@@ -123,10 +123,10 @@ def unlink(self):
 
     def toggle_active(self):
         super(SaleCouponProgram, self).toggle_active()
-        if not self.active:
-            coupons = self.filtered(lambda p: p.promo_code_usage == 'code_needed').mapped('coupon_ids')
-            coupons.filtered(lambda x: x.state != 'used').write({'state': 'expired'})
-        self.mapped('discount_line_product_id').write({'active': self.active})
+        for program in self:
+                program.discount_line_product_id.active = program.active
```

Comment:
I don't want to sound annoying but there is an extra tab here it looks

Reply:
:+1: for notice.
 Fixed